client/sites/marketing/tools/index.tsx

```diff
@@ -20,21 +20,34 @@ import { getMarketingFeaturesData } from './marketing-features-data';
 
 import './style.scss';
 
+const items = [
+	{ key: '', text: 'All' },
+	{ key: 'new', text: 'New tools' },
+	{ key: 'favourite', text: 'User favorites' },
+];
+
 export default function MarketingTools() {
 	const translate = useTranslate();
 	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const [ selectedCategory, setSelectedCategory ] = useState( '' );
 	const selectedSiteSlug: T.SiteSlug | null = useSelector( getSelectedSiteSlug );
 	const siteId = useSelector( getSelectedSiteId ) || 0;
 
 	const marketingFeatures = getMarketingFeaturesData( selectedSiteSlug, translate, localizeUrl );
 
 	const marketingFeaturesFiltered = useMemo( () => {
-		return marketingFeatures.filter(
-			( feature ) =>
-				feature.title.toLowerCase().includes( searchTerm.toLowerCase() ) ||
-				feature.description.toLowerCase().includes( searchTerm.toLowerCase() )
+		let filteredFeatures = marketingFeatures.filter( ( feature ) =>
+			( feature.title.toLowerCase() + feature.description.toLowerCase() ).includes(
+				searchTerm.toLowerCase()
+			)
 		);
-	}, [ searchTerm, marketingFeatures ] );
+		if ( selectedCategory ) {
+			filteredFeatures = filteredFeatures.filter( ( feature ) =>
+				feature.categories.includes( selectedCategory )
+			);
+		}
```

Comment:
I think we should swap the precedence of categories and search here, it'd make sense to remove the category filtering when we have a search term. Otherwise, the category filter can hide potential results for the search:

https://github.com/user-attachments/assets/2e76b456-bbb8-461a-8915-2f4d09f6d145

Reply:
Do you mean a behaviour similar to the [plugins search](https://wordpress.com/plugins), right? Clearing the search term when a category is selected and clearing the category when a search term is inserted?

https://github.com/user-attachments/assets/dcad7cb5-1c6c-41e2-9d74-1e1eafa700fc

Reply:
Yes, exactly! 🙌 I couldn't add a suggestion because the changes included deleted lines, and then GH wouldn't let me add one, but I think something like this would do the trick regarding the filtering logic:
```
		let filteredFeatures = marketingFeatures.filter( ( feature ) =>
			feature.categories.includes( selectedCategory )
		);
		if ( searchTerm ) {
			filteredFeatures = filteredFeatures.filter( ( feature ) =>
				( feature.title.toLowerCase() + feature.description.toLowerCase() ).includes(
				        searchTerm.toLowerCase()
			         )
			);
		}
```

But the filter should still be reset so that the `All` option is highlighted.

Reply:
In this case, both components, the `Search` and `ResponsiveToolbarGroup` are uncontrolled components so I will need to take a reference to them to update their initial state. It should not be a big deal but it will be slightly more complex than changing the order of filtering, I will suggest the changes and we can iterate over them.